src/cunumeric/index/advanced_indexing.cu

```diff
@@ -104,12 +104,10 @@ struct AdvancedIndexingImplBody<VariantKind::GPU, CODE, DIM, OUT_TYPE> {
       count_nonzero_kernel<<<blocks, THREADS_PER_BLOCK, 0, stream>>>(
         volume, size, offsets, in, pitches, rect.lo, 1, skip_size, key_dim);
 
-    cudaStreamSynchronize(stream);
-
     auto off_ptr = offsets.ptr(0);
     thrust::exclusive_scan(thrust::cuda::par.on(stream), off_ptr, off_ptr + volume, off_ptr);
 
```

Comment:
Should there be a CHECK_CUDA_STREAM here?

Reply:
[9383c2a](https://github.com/nv-legate/cunumeric/pull/509/commits/9383c2a770864728fd77494a713ae4ecf62e8e12)